lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -214,11 +214,12 @@ sub to_idn {
     my ( $self, $str ) = @_;
 
     if ( $str =~ m/^[[:ascii:]]+$/ ) {
+        $str = lc $str;
         return $str;
     }
 
     if ( Zonemaster::LDNS::has_idn() ) {
-        return Zonemaster::LDNS::to_idn( $str );
+        return lc Zonemaster::LDNS::to_idn( $str );
```

Comment:
The A-label will always match /^xn--[a-z0-9-]+$/ by definition. (The dot is outside the label.)